packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -209,7 +209,17 @@ function translateTypeAnnotation(
           return emitRootTag(nullable);
         }
         case 'Promise': {
-          return emitPromise(hasteModuleName, typeAnnotation, parser, nullable);
+          return emitPromise(
+            hasteModuleName,
+            typeAnnotation,
+            parser,
+            nullable,
+            types,
+            aliasMap,
+            tryParse,
+            cxxOnly,
+            translateTypeAnnotation,
+          );
```

Comment:
Here I would call the `translateTypeAnnotation` to get the `elementType`.
It should make the APIs more composable, what do you think?

Reply:
1) answered above, for short it is `*` in `Promise<*>`

2) the generic argument is checked inside `emitPromise`, it is better to deal with it inside `emitPromise`. A shorter function might just "look" better, but it is complex because the requirement is complex. So I would like to keep it in this way.

And this abstraction is just good enough (like `emitFunction`), since `emitPromise` never knows about the differences between languages, and it still able to do recursion, as AST itself is recursive.